### DISCUSSION TITLE
Question about .aab

### DISCUSSION THREAD

**Discussion by Mintonist** [airsdk/Adobe-Runtime-Support#1605] [General]
Hi!
What is recommendation config in adt.cfg for .aab building?

I notice size difference with parameter:
UncompressedExtensions=emd,tfl,tflite,pb

![image](https://user-images.githubusercontent.com/9267850/151801959-1ed7b7f0-4585-4b6b-be02-a04c1bc310e9.png)

In `/base` directory all .pb files are the same (no CRC difference)
![image](https://user-images.githubusercontent.com/9267850/151806473-7856984f-1892-4c1b-b315-6ec0e29c57ba.png)

I have no .pb files in my .apk  so I have not paid attention early

Something else important?)
Thx!

**Comment by ajwfrost** [comment]
The uncompressed extensions listed in that config file are used for the options we pass to `aapt`, for packaging up the APK. But it wouldn't actually impact those .pb files in the root/base folder, it would mostly impact the included asset files plus some resources.

If you look at these files in a zip manager, you'll see that some are listed as "Store" and some are "Deflate", i.e. they are compressed when they're included. By default e.g. SWF files and XML files are compressed, png files are just stored. If you use the 'uncompressed' list, then in the "assets" folder where your main application binaries are stored, you can prevent the "Deflate" of some files. Likewise it appears to affect the files in /res/raw, but it seems other XML files (generated?) are still set to use "Deflate"...

You should notice the different more if you use:
```
UncompressedExtensions=swf
```

thanks

**Reply by Mintonist** [reply]
Thx!
Is any idea about different size of BundleConfig.pb with and without UncompressedExtensions=pb ?

**Reply by ajwfrost** [reply]
The BundleConfig.pb file is a private format from Android so not entirely sure what's in it or why there's a difference. They are stored 'compressed' by the looks of things. I would guess that they have some kind of index of what else is in the bundle so a difference elsewhere in the bundle structure may result in a difference here, and a minor change to the file could make it more compressable hence a difference in size..? But that's a lot of conjecture...

thanks
